### DISCUSSION TITLE
Migrating from x86_64 to arm64 Homebrew

### DISCUSSION THREAD

**Discussion by JordanField** [Homebrew/discussions#437] [Getting started] [answered]
Hi all, 

Fantastic work getting Apple Silicon working!
Apologies if there's a guide for this somewhere, but I haven't found anything.

I want to:

1. Install native arm64 `brew`
2. Collect all formulae (but _not_ casks) installed via x86_64 `brew` (henceforth `ibrew`, since that's the alias I use to run using Rosetta)
3. Re-install these formula using `brew`
4. Remove the Intel formulae
5. Remove `ibrew` entirely.

What is the least disruptive way to do this? Thanks in advance 😄

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-bundle

**Comment by tail-call** [comment]
Hi everyone! I suggest referring to this post for migration tips:

https://blog.smittytone.net/2021/02/07/how-to-migrate-to-native-homebrew-on-an-m1-mac/

Also, like @SMillerDev pointed out, `brew bundle dump` for creating a Brewfile and then using `brew bundle` on a new installation is very convenient.

**Comment by ykorzikowski** [comment]
Migration from homebrew x86 to arm64

- Old Homebrew location: /usr/local/Homebrew
- New Homebrew location: /opt/homebrew

```
sudo mv /usr/local/Homebrew /usr/local/Homebrew.x86

/usr/local/Homebrew.x86/bin/brew bundle dump
/opt/homebrew/bin/brew bundle

ls usr/local/Homebrew.x86/Cellar|xargs /opt/homebrew/bin/brew install $1
ls usr/local/Homebrew.x86/Caskroom|xargs /opt/homebrew/bin/brew install --cask $1 --force

rm -rf Homebrew.x86/
```

**Reply by SMillerDev** [reply, marked as answer]
Most of these steps are not needed. If you already have homebrew installed in both locations, this is all you need.

```
arch -x86_64 /usr/local/Homebrew/bin/brew bundle dump
arch -arm64 /opt/homebrew/bin/brew bundle
```
